Developer-Guide.md

```diff
@@ -0,0 +1,193 @@
+* [Warning](#warning)
+* [Assumptions](#assumptions)
+* [Build and install a Kata Containers runtime](#build-and-install-a-kata-containers-runtime)
+    * [Check hardware requirements](#check-hardware-requirements)
+    * [Enable full debug](#enable-full-debug)
+* [Build and install Kata proxy](#build-and-install-kata-proxy)
+* [Build and install Kata shim](#build-and-install-kata-shim)
+* [Create an image](#create-an-image)
+    * [Build a custom Kata agent - OPTIONAL](#build-a-custom-kata-agent---optional)
+    * [Get the osbuilder](#get-the-osbuilder)
+    * [Create a rootfs image](#create-a-rootfs-image)
+        * [Add a custom agent to the image - OPTIONAL](#add-a-custom-agent-to-the-image---optional)
+    * [Build the image](#build-the-image)
+* [Install the image](#install-the-image)
+* [Install guest kernel images](#install-guest-kernel-images)
+* [Update Docker configuration](#update-docker-configuration)
+* [Create a Kata Container](#create-a-kata-container)
+* [Problem determination](#problem-determination)
+* [Appendices](#appendices)
+    * [Checking Docker default runtime](#checking-docker-default-runtime)
+
+# Warning
+
+This document is written **specifically for developers**.
+
+# Assumptions
+
+- You are working on a non-critical test or development system.
+- You already have the following installed:
+  - [Docker](https://www.docker.com/).
+  - [golang](https://golang.org/dl) version 1.8.3 or newer.
+  - `make`.
+  - `gcc` (required for building the shim and runtime).
+
+- You have installed the Clear Containers `linux-container` and `qemu-lite`
+  packages containing the guest kernel images and hypervisor. These packages
+  are automatically installed when you install Clear Containers, but can be
+  installed separately:
+
+    https://github.com/clearcontainers/runtime/wiki/Installation
+
+# Build and install the Kata Containers runtime
+
+```
+$ go get -d -u github.com/kata-containers/runtime
+$ cd $GOPATH/src/github.com/kata-containers/runtime
+$ make && sudo -E PATH=$PATH make install
+```
+
+The build will create:
+
+- runtime binary: `/usr/local/bin/kata-runtime`
+- configuration file: `/usr/share/defaults/kata-containers/configuration.toml`
+
+# Check hardware requirements
+
+You can now check if your system is capable of creating a Kata Container by running:
+
+```
+$ sudo kata-runtime kata-check
+```
+
+If your system is *not* able to run Kata Containers, this command will error and explain why.
```

Comment:
If your system is *not* able to run Kata Containers, the previous command will error and explain why.